src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseDataType.java

```diff
@@ -6,30 +6,49 @@
  */
 public enum ClickHouseDataType {
     // Unsigned
-    UInt8,
-    UInt16,
-    UInt32,
-    UInt64,
+    UInt8("UInt8"),
+    UInt16("UInt16"),
+    UInt32("UInt32"),
+    UInt64("UInt64"),
 
     // Signed
-    Int8,
-    Int16,
-    Int32,
-    Int64,
+    Int8("Int8"),
+    Int16("Int16"),
+    Int32("Int32"),
+    Int64("Int64"),
 
     // Floating point
-    Float32,
-    Float64,
+    Float32("Float32"),
+    Float64("Float64"),
 
     // Date time
-    DateTime,
-    Date,
+    DateTime("DateTime"),
+    Date("Date"),
+
+    Decimal("Decimal"),
 
     // Misc
-    String;
+    String("String");
+
+    String typo;
 
-    public String getName(boolean isNullable) {
-        String retval = toString();
+    ClickHouseDataType(String string) {
```

Comment:
Maybe, let's have 2 constructors? No-argument calls single-argument with this.name().

Reply:
no-arguments constructor was added